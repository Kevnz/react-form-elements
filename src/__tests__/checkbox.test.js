import React from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import Checkbox from '../checkbox'

const LABEL = 'My Checkbox Label'
describe('Checkbox component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { getByText, container } = render(<Checkbox label={LABEL} />)

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText(LABEL)
      expect(label.textContent).toBe(LABEL)
    })
  })
  it('should have checked attribute', () => {
    const { getByLabelText, container } = render(
      <Checkbox label={LABEL} name="TheName" checked />
    )

    expect(container.firstChild).toMatchSnapshot()
    const check = getByLabelText(LABEL)
    expect(check.checked).toBe(true)
  })
  it('should not have checked attribute', () => {
    const { getByLabelText, container } = render(
      <Checkbox label={LABEL} name="TheName" />
    )

    expect(container.firstChild).toMatchSnapshot()
    const check = getByLabelText(LABEL)
    expect(check).toMatchSnapshot()
    expect(check.checked).toBe(false)
  })
})
describe('handling input', () => {
  it('should set a value when passed in', () => {
    const { getByLabelText, container } = render(
      <Checkbox label="My Label" initialValue="The Value" checked />
    )

    expect(container.firstChild).toMatchSnapshot()
    const input = getByLabelText('My Label')
    expect(input.checked).toBe(true)
  })
  it('should update a value when changed in', () => {
    const { getByLabelText, container } = render(
      <Checkbox label="My Label" initialValue="The Value" />
    )

    expect(container.firstChild).toMatchSnapshot()
    const input = getByLabelText('My Label')
    fireEvent.change(input, { target: { value: 43 } })
    expect(input.value).toBe('43')
  })
})
