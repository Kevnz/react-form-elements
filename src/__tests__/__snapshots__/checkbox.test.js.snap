// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-1-1"
  >
    My Checkbox Label
  </label>
  <input
    id="id-1-1"
    type="checkbox"
    value=""
  />
</div>
`;

exports[`Checkbox component handling input should set a value when passed in 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-2-1"
  >
    My Label
  </label>
  <input
    checked=""
    id="id-2-1"
    type="checkbox"
    value="test value"
  />
</div>
`;

exports[`Checkbox component handling input should update a value when changed in 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-3-1"
  >
    My Label
  </label>
  <input
    id="id-3-1"
    type="checkbox"
    value="updateval"
  />
</div>
`;
